pkg/controller/launch/pool_manager.go

```diff
@@ -109,23 +133,46 @@ func (cpm *ConcretePoolManager) SetStatus(status *PoolStatus) error {
 
 	//TODO: Use util.UpdateKlusterWithRetries here
 	copy, err := cpm.Clients.Kubernikus.Kubernikus().Klusters(cpm.Kluster.Namespace).Get(cpm.Kluster.Name, metav1.GetOptions{})
+
 	if err != nil {
 		return err
 	}
 
-	for i, curInfo := range copy.Status.NodePools {
-		if curInfo.Name == newInfo.Name {
-			if curInfo == newInfo {
-				return nil
+	updated := false
+	// Add new pools in the spec to the status
+	for _, specPool := range copy.Spec.NodePools {
+		// Find the pool
+		if npi, ok := nodePoolInfoGet(copy.Status.NodePools, specPool.Name); ok {
+			// is there a need to update?
```

Comment:
I like it better if the comment describes the `if` case: e.g. "Skip pool if nothing changed"